nautobot/docs/development/docker-compose-advanced-use-cases.md

```diff
@@ -0,0 +1,121 @@
+# Docker Compose Advanced Use Cases
+
+This section describes some of the more advanced use cases for the [Docker Compose development workflow](getting-started.md#docker-compose-workflow).
+
+## Invoke Configuration
+
+The Invoke tasks have some default [configuration](http://docs.pyinvoke.org/en/stable/concepts/configuration.html) which you may want to override. Configuration properties include:
+
+- `project_name`: The name that all Docker containers will be grouped together under (default: `nautobot`, resulting in containers named `nautobot_nautobot_1`, `nautobot_redis_1`, etc.)
+- `python_ver`: the Python version which is used to build the Docker container (default: `3.7`)
+- `local`: run the commands in the local environment vs the Docker container (default: `False`)
+- `compose_dir`: the full path to the directory containing the Docker Compose YAML files (default: `"<nautobot source directory>/development"`)
+- `compose_files`: the Docker Compose YAML file(s) to use (default: `["docker-compose.yml", "docker-compose.dev.yml"]`)
+- `docker_image_names_main` and `docker_image_names_develop`: Used when [building Docker images for publication](release-checklist.md#publish-docker-images); you shouldn't generally need to change these.
+
+These setting may be overridden several different ways (from highest to lowest precedence):
+
+- Command line argument on the individual commands (see `invoke $command --help`) if available
+- Using environment variables such as `INVOKE_NAUTOBOT_PYTHON_VER`; the variables are prefixed with `INVOKE_NAUTOBOT_` and must be uppercase; note that Invoke does not presently support environment variable overriding of list properties such as `compose_files`.
+- Using an `invoke.yml` file (see [`invoke.yml.example`](https://github.com/nautobot/nautobot/blob/main/invoke.yml.example))
+
+## Working with Docker Compose
+
+The files related to the Docker development environment can be found inside of the `development` directory at the root of the project.
+
+In this directory you'll find the following core files:
+
+- `docker-compose.build.yml` - Docker compose override file used to start/build the production docker images for local testing.
+- `docker-compose.debug.yml` - Docker compose override file used to start the Nautobot container for use with [Visual Studio Code's dev container integration](#microsoft-visual-studio-code-integration).
+- `docker-compose.dev.yml` - Docker compose override file used to mount the Nautobot source code inside the container at `/source` and the `nautobot_config.py` from the same directory as `/opt/nautobot/nautobot_config.py` for the active configuration.
+- `docker-compose.yml` - Docker service containers and their relationships to the Nautobot container
```

Comment:
Seems like maybe the base `docker-compose.yml` should be listed first, before the various override files?